StyleCop.Analyzers/StyleCop.Analyzers/Settings/SettingsHelper.cs

```diff
@@ -57,5 +66,51 @@ private static StyleCopSettings GetStyleCopSettings(ImmutableArray<AdditionalTex
 
             return new StyleCopSettings();
         }
+
+        /// <summary>
+        /// This code works around dotnet/roslyn#6596 by using the file system APIs instead of the Roslyn APIs to read
+        /// the additional text. If the file system APIs are not available, the code falls back to the previous
+        /// behavior.
+        /// </summary>
+        /// <param name="additionalText">The additional text to read.</param>
+        /// <param name="cancellationToken">The cancellation token that the operation will observe.</param>
+        /// <returns>The content of the additional text as a string.</returns>
+        private static string ReadAdditionalText(AdditionalText additionalText, CancellationToken cancellationToken)
+        {
+            if (fileExists == null)
+            {
+                Type fileClass = typeof(string).GetTypeInfo().Assembly.GetType("System.IO.File");
+                if (fileClass != null)
+                {
+                    MethodInfo readAllText = fileClass.GetRuntimeMethod("ReadAllText", new[] { typeof(string) });
+                    MethodInfo exists = fileClass.GetRuntimeMethod("Exists", new[] { typeof(string) });
+                    if (readAllText != null && exists != null)
+                    {
+                        Interlocked.CompareExchange(ref fileReadAllText, (Func<string, string>)readAllText.CreateDelegate(typeof(Func<string, string>)), null);
+                        Interlocked.CompareExchange(ref fileExists, (Func<string, bool>)exists.CreateDelegate(typeof(Func<string, bool>)), null);
+                    }
+                }
+
+                if (fileExists == null)
+                {
+                    // this special case allows for a clean fall back to AdditionalText.GetText()
+                    fileExists = _ => false;
+                }
+            }
+
+            try
+            {
+                if (fileExists(additionalText.Path))
+                {
+                    return fileReadAllText(additionalText.Path);
+                }
+            }
+            catch (IOException)
```

Comment:
This can throw [all sorts of exceptions](https://msdn.microsoft.com/en-us/library/ms143368%28v=vs.110%29.aspx#Anchor_1). You'll want to catch them all.